codegen/type_converter.go

```diff
@@ -638,25 +689,61 @@ func (c *TypeConverter) genStructConverter(
 			var (
 				stFromPrefix = keyPrefix
 				stFromType   compile.TypeSpec
+				fromTypeName string
 			)
 			if fromField != nil {
 				stFromType = fromField.Type
 				stFromPrefix = keyPrefix + pascalCase(fromField.Name)
+
+				var err error
+				fromTypeName, err = c.getIdentifierName(stFromType)
+				if err != nil {
+					fromTypeName = ""
+				}
 			}
-			err := c.genConverterForStruct(
-				toField.Name,
-				toFieldType,
-				toField.Required,
-				stFromType,
-				fromIdentifier,
-				keyPrefix+pascalCase(toField.Name),
-				stFromPrefix,
-				indent,
-				fieldMap,
-			)
+
+			toTypeName, err := c.getIdentifierName(toFieldType)
 			if err != nil {
 				return err
 			}
+
+			if converterMethodName, ok := c.convStructMap[toFieldType.Name]; ok {
+				// the converter for this struct has already been generated, so just use it
+				c.append(indent, "out.", keyPrefix+pascalCase(toField.Name), " = ", converterMethodName, "(", fromIdentifier, ")")
+			} else if c.useRecurGen && fromTypeName != "" {
```

Comment:
I always prefer `len(fromTypeName) < 1`, string len compare is cheaper than string compare

Reply:
I hope the compiler is magical enough to save me😅